Bring RK3566-X55 up to speed with mainline

ananjaser1211 opened the pull request:
## Description

Note : This is an initial PR to bring the RK3566-X55 device some what in line with other devices, this is W.I.P for the most part as im still working my way through JELOS

- This includes changes and patches needed to compile and boot RK3566-X55 Device on latest jelos dev branch
- Also includes patches by @Sparticuz to fix WiFi SDIO issues (WiFi is now working pre/after sleep)
- Includes SWAY integration from @sydarn 
- Symlink platform quirks to RK3566 as RK3566-X55 had no quirks implemented
- Use controller configs from RK3566-BSP-X55 for retroarch etc (Requires review https://github.com/JustEnoughLinuxOS/distribution/commit/31a18b393fb0a62eb81aef2daf2eeed50de0dcd9)

## Current Issues on RK3566-X55
- ~EmulationStation > Tools Menu is non functional (PortMaster, Filemanager, Retroarch etc does not lunch)~
- ~Retroarch menu (L3/R3 Ingame) is zoomed in and is not displaying at 720p rather a smaller resolution~
- mupenn64plus-sa does not detect input (need to try a clean flash)
- Brightness slider can be set to 0 (display turns off)
- GAMMA/CONTRAST/SATURATION/HUE can't be adjusted

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested Locally?

- [X] Test A
- [ ] Test B

- Successful Compilation
- boot into emulation station
- run games through RetroArch and PPSSPP
- Test Volume/Volume Change
- Test Sleep functionality
- Test WiFi connectivity & SSH
- Test Brightness change
- Test Device LED options

**Test Configuration**:
* Build OS name and version: Linux Mint 21 (22.04)
* Docker (Y/N): N
* JELOS Branch: dev
* Any additional information that may be useful:

## Checklist:

- [X] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

Sparticuz commented:
https://github.com/JustEnoughLinuxOS/distribution/commit/a820c6162c04f7b7ec781cfa0dc70db28c594677 is not needed, just make sure the correct config is set in the kernel config. I think it's `CONFIG_RTW88_8821CE`, but I'm on mobile right now and can't easily check. You can look in the rk3566 kernel config.

ananjaser1211 commented:
- Dropped https://github.com/JustEnoughLinuxOS/distribution/commit/a820c6162c04f7b7ec781cfa0dc70db28c594677
- Dropped `projects/Rockchip/packages/u-boot/patches/RK3566-X55/003-fix-dtb-and-vs.patch` and `projects/Rockchip/packages/u-boot/patches/RK3566-X55/004-rgb10max3.patch` U-BOOT patches
- added back `packages/emulators/standalone/retroarch/patches/RK3566-X55/003-fix-go2-rotation.patch` since it was fine

ananjaser1211 commented:
* Dropped dolphin configs
* fixed up emulator configs
* fixed retroarch scaling issues
* disable the input udev rules (they caused some weird behavior with d-pad up/down being sticky)
* added 1.9ghz cpu clock (did not work in turbo mode, so i added it by default like in the bsp)

Remaining issues :
- Display settings is not working (i tried changing the connector to 55 which is what on my device, no avail)
- mupenn64plus-sa does not detect input (need to try a clean flash)
- ppsspp stopped working as of now (Failed to initialize glew) will re-try with a clean build
- i think CPU utilization is too high during video screen saver, as the device gets quite warm

Sparticuz commented on packages/graphics/mesa/package.mk:
```diff
@@ -17,7 +17,7 @@ case ${DEVICE} in
   ;;
   # keep RK3399 and RK-ARMV8-A on same version to maintain cross comaptibility
-  RK33*|RK-ARMV8-A|RK3566) #Using upstream dev for panfrost
+  RK33*|RK-ARMV8-A|RK3566*) #Using upstream dev for panfrost
```
I think this might also grab RK3566-BSP, may need to specify RK3566-X55 instead
